File: OsmAnd/src/net/osmand/plus/myplaces/tracks/filters/SmartFolderUpdateListener.java
package net.osmand.plus.myplaces.tracks.filters;

import androidx.annotation.WorkerThread;

import net.osmand.plus.track.data.SmartFolder;

public interface SmartFolderUpdateListener {
	default void onSmartFoldersUpdated() {
	}

	@WorkerThread
	default void onSmartFolderUpdated(SmartFolder smartFolder) {
	}

	default void onSmartFolderRenamed(SmartFolder smartFolder) {
	}

	default void onSmartFolderSaved(SmartFolder smartFolder) {
	}

	default void onSmartFolderCreated(SmartFolder smartFolder) {
	}
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/filters/SmartFolderUpdateListener.kt
package net.osmand.shared.filters

import androidx.annotation.WorkerThread

interface SmartFolderUpdateListener {
	fun onSmartFoldersUpdated()

	@WorkerThread
	fun onSmartFolderUpdated(smartFolder: SmartFolder)
	fun onSmartFolderRenamed(smartFolder: SmartFolder)
	fun onSmartFolderSaved(smartFolder: SmartFolder)
	fun onSmartFolderCreated(smartFolder: SmartFolder)
}
